views/components/ConfigurationCard/index.jsx

```diff
@@ -0,0 +1,187 @@
+import React from 'react'
+import moment from 'moment'
+import {
+  Card,
+  CardHeader,
+  Divider,
+  Typography,
+  CardActions,
+  Switch,
+  IconButton,
+  FormControlLabel,
+} from '@material-ui/core'
+import { Edit, Clear, Share } from '@material-ui/icons'
+import FullView from '@material-ui/icons/AspectRatio'
+import Copy from '@material-ui/icons/FileCopy'
+import ConfigCardAvatar from '../ConfigurationCardAvatar'
+import openNewWindow from '../../fe-utils/windowUtil'
+import { UserModel, UserConfigurationsModel } from '../../models'
+import { routes } from '../../routes/routes'
+import { colors } from '../../../constants'
+
+const ConfigurationCard = ({
+  classes,
+  openSearch,
+  loadAllConfigurations,
+  user,
+  config,
+  preferences,
+  setPreferences,
+  setConfigurations,
+  width,
+}) => {
+  const { uid } = user
+  const { _id, name, owner, readers, type } = config
+  const ownsConfig = uid === owner
+  const showTime = config.modified || config.created
+  const localTime = moment.utc(showTime).local().format()
+  const updated = moment(localTime).calendar()
+  const checked = config._id === preferences.config
+
+  const copyConfig = async (configuration) => {
```

Comment:
we should be passing these handler functions as props to this component. So there should be an `onCopyConfig`, `onRemoveConfig`, etc prop passed here so we won't need to call the models or set parent state directly in this component.

Reply:
[d31c716](https://github.com/AMP-SCZ/dpdash/pull/310/commits/d31c716b584e340b90068954fc940c49761e52fe)